include/swift/AST/TypeRepr.h

```diff
@@ -476,8 +476,11 @@ class FunctionTypeRepr : public TypeRepr {
   // we can have polymorphic and substituted function values.
```

Comment:
IIUC, this comment applies to both the first and second group of fields (out of 3 groups total, with the middle one introduce in this PR), but proximity seems to indicate that it only applies to the first set of fields. Could you move things around or add an extra comment like:

```
// These fields are applicable for both AST and SIL.
```

between the second group and third group so it becomes clear the above comment also applies to the second group.